src/storage.rs

```diff
@@ -1,61 +1,226 @@
 use crate::types::{Stories, Story};
 use log::{error, info};
+use rusqlite::Connection;
 use std::error::Error;
+use std::sync::Arc;
 use tokio::fs;
+use tokio::sync::Mutex;
 
-const STORAGE_FILE_PATH: &str = "./stories.json";
-const PEER_NAME_FILE_PATH: &str = "./peer_name.json";
+const DATABASE_PATH: &str = "./stories.db";
+const PEER_NAME_FILE_PATH: &str = "./peer_name.json"; // Keep for backward compatibility
+
+// Thread-safe database connection
+static DB_CONN: once_cell::sync::OnceCell<Arc<Mutex<Connection>>> = once_cell::sync::OnceCell::new();
+
+async fn get_db_connection() -> Result<Arc<Mutex<Connection>>, Box<dyn Error>> {
+    if let Some(conn) = DB_CONN.get() {
+        Ok(conn.clone())
+    } else {
+        info!("Creating new SQLite database connection: {}", DATABASE_PATH);
+        
+        // Create the database file and connection
+        let conn = Connection::open(DATABASE_PATH)?;
+        let conn_arc = Arc::new(Mutex::new(conn));
+        
+        // Initialize the connection in the static variable
+        DB_CONN.set(conn_arc.clone()).map_err(|_| "Failed to initialize database connection")?;
+        
+        info!("Successfully connected to SQLite database");
+        Ok(conn_arc)
+    }
+}
+
+async fn create_tables() -> Result<(), Box<dyn Error>> {
```

Comment:
Would it be possible for the migrations to be done outside of the storage.rs file.  Can the migrations be moved to a separate file?